Stop creating unnecessary Storage with newWithStorage1d.

gchanan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #34403 Port set_ to ATen.
* **#34389 Stop creating unnecessary Storage with newWithStorage1d.**
* #34388 Kill newWithStorage.
* #34387 Get rid of newWithSize.

Differential Revision: [D20311060](https://our.internmc.facebook.com/intern/diff/D20311060)

gchanan committed:
Stop creating unnecessary Storage with newWithStorage1d.

gchanan committed:
Update on "Stop creating unnecessary Storage with newWithStorage1d."


Differential Revision: [D20311060](https://our.internmc.facebook.com/intern/diff/D20311060)

gchanan committed:
Update on "Stop creating unnecessary Storage with newWithStorage1d."


Differential Revision: [D20311060](https://our.internmc.facebook.com/intern/diff/D20311060)

gchanan committed:
Update on "Stop creating unnecessary Storage with newWithStorage1d."


Differential Revision: [D20311060](https://our.internmc.facebook.com/intern/diff/D20311060)

gchanan committed:
Update on "Stop creating unnecessary Storage with newWithStorage1d."


Differential Revision: [D20311060](https://our.internmc.facebook.com/intern/diff/D20311060)

gchanan committed:
Update on "Stop creating unnecessary Storage with newWithStorage1d."


Differential Revision: [D20311060](https://our.internmc.facebook.com/intern/diff/D20311060)

bhosmer reviewed (approved):
I think this explains what I was seeing in #34382 :)

gchanan committed:
Update on "Stop creating unnecessary Storage with newWithStorage1d."


Differential Revision: [D20311060](https://our.internmc.facebook.com/intern/diff/D20311060)

gchanan committed:
Update on "Stop creating unnecessary Storage with newWithStorage1d."


Differential Revision: [D20311060](https://our.internmc.facebook.com/intern/diff/D20311060)

gchanan committed:
Update on "Stop creating unnecessary Storage with newWithStorage1d."


Differential Revision: [D20311060](https://our.internmc.facebook.com/intern/diff/D20311060)

gchanan committed:
Update on "Stop creating unnecessary Storage with newWithStorage1d."


Differential Revision: [D20311060](https://our.internmc.facebook.com/intern/diff/D20311060)

gchanan committed:
Update on "Stop creating unnecessary Storage with newWithStorage1d."


Differential Revision: [D20311060](https://our.internmc.facebook.com/intern/diff/D20311060)
